test/GroupByTest.cpp

```diff
@@ -1588,14 +1576,13 @@ TEST_F(GroupByOptimizations, computeGroupByForSingleIndexScan) {
   // `computeGroupByForJoinWithFullScan` method", `chooseInterface == false`
   // means use the general `computeOptimizedGroupByIfPossible` function.
   auto testWithBothInterfaces = [&](bool chooseInterface) {
-    IdTable result{qec->getAllocator()};
     auto groupBy =
         GroupBy{qec, emptyVariables, aliasesCountX, xyzScanSortedByX};
-    if (chooseInterface) {
-      ASSERT_TRUE(groupBy.computeGroupByForSingleIndexScan(&result));
-    } else {
-      ASSERT_TRUE(groupBy.computeOptimizedGroupByIfPossible(&result));
-    }
+    auto optional = chooseInterface
+                        ? groupBy.computeGroupByForSingleIndexScan()
+                        : groupBy.computeOptimizedGroupByIfPossible();
+    ASSERT_TRUE(optional.has_value());
+    IdTable result = std::move(optional).value();
```

Comment:
You can also simplify these result checks by the helper function `makeIdTableFromVector` 
(clean up as you go)